schema.go

```diff
@@ -40,7 +40,7 @@ type Schema struct {
 	AdditionalProperties *Schema                                 `json:"additionalProperties,omitempty"` // section 10.3.2.3
 	PropertyNames        *Schema                                 `json:"propertyNames,omitempty"`        // section 10.3.2.4
 	// RFC draft-bhutton-json-schema-validation-00, section 6
-	Type              string              `json:"type,omitempty"`              // section 6.1.1
+	Type              []string            `json:"type,omitempty"`              // section 6.1.1
```

Comment:
So this is a significant breaking change for anyone using the `Type` field directly, and as such, I'm not convinced this is the correct approach.

I'm wondering if an alternative approach, given how rare this field is even used, might be to automatically handle a comma or semicolon separated list of types alongside the automatic conversion in the JSON marshalling methods.

A definition would effectively look like:

```
jsonschema.Schema{
  Type: "string,null",
}
```

or in a struct:

```
type Object struct {
  Text string `jsonschema:"type=string,null"`
}
```

Reply:
For the structure, we could even get it from the property type or the json tag. If it is a pointer or the json tag has the `omitempty` the `null` could be added to the jsonchema array